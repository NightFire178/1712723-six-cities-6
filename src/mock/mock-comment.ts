import mockUser from "./mock-user";
import comment from '../types/comment'

const mockComment:comment = {
  comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
  date: `2021-03-07T08:04:28.647Z`,
  id: 10,
  rating: 4,
  user: mockUser
}
export default mockComment

export const mockArrayComments:Array<comment> =[
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 15,
    rating: 3,
    user: mockUser
  },
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 13,
    rating: 5,
    user: mockUser
  },
  {
    comment: `What an amazing view! The house is stunning and in
    an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 10,
    rating: 4,
    user: mockUser
  }
]
