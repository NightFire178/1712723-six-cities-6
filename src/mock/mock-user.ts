import User from "../types/user";

const mockUser:User = {
  id: 1,
  name: `vasa`,
  is_pro: true,
  avatar_url:`https://assets.htmlacademy.ruâ€¦tar/6.jp`,
  email: `sdfhskd@maid.ru`
}
export default mockUser
