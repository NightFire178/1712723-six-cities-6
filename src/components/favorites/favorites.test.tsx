import React from 'react'
import {render} from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import Favorites from "./favorites";
import {mockArrayHotels} from "../../mock/mock-hotels";

const useFavoriteSelection = {
  hotelsFavorite: ()=>JSON.parse(JSON.stringify(mockArrayHotels))
}
jest.mock("../../hooks/use-selectors-state/use-favorite", ()=> useFavoriteSelection)

//eslint-disable-next-line
jest.mock("../block/header/header", ()=>()=><div>Header</div>)

//eslint-disable-next-line
jest.mock("../block/card/card", ()=>()=><div>Card</div>)

const thunkFavorites = ()=>true
jest.mock('../../hooks/use-thunk',() =>()=>({thunkFavorites}))

test(`favorites component`, () => {
  const history = createMemoryHistory()
  const favorites = render(
    <Router history={history}>
      <Favorites/>
    </Router>
  )
  expect(favorites).toMatchSnapshot()
})
