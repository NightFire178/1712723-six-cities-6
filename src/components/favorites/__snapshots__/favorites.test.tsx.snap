// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favorites component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="display: none;"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
          >
            <symbol
              id="icon-arrow-select"
              viewBox="0 0 7 4"
            >
              <path
                clip-rule="evenodd"
                d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
                fill-rule="evenodd"
              />
            </symbol>
            <symbol
              id="icon-bookmark"
              viewBox="0 0 17 18"
            >
              <path
                d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
              />
            </symbol>
            <symbol
              id="icon-star"
              viewBox="0 0 13 12"
            >
              <path
                clip-rule="evenodd"
                d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
                fill-rule="evenodd"
              />
            </symbol>
          </svg>
        </div>
        <div
          class="page"
        >
          <div>
            Header
          </div>
          <main
            class="page__main page__main--favorites page__main--favorites-empty"
          >
            <div
              class="page__favorites-container container"
            >
              <section
                class="favorites favorites--empty"
              >
                <h1
                  class="visually-hidden"
                >
                  Favorites (empty)
                </h1>
                <div
                  class="favorites__status-wrapper"
                >
                  <b
                    class="favorites__status"
                  >
                    Nothing yet saved.
                  </b>
                  <p
                    class="favorites__status-description"
                  >
                    Save properties to narrow down search or plan your future trips.
                  </p>
                </div>
              </section>
            </div>
          </main>
          <footer
            class="footer container"
          >
            <a
              href="/"
            >
              <div
                class="footer__logo-link"
              >
                <img
                  alt="6 cities logo"
                  class="footer__logo"
                  height="33"
                  src="img/logo.svg"
                  width="64"
                />
              </div>
            </a>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="display: none;"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        >
          <symbol
            id="icon-arrow-select"
            viewBox="0 0 7 4"
          >
            <path
              clip-rule="evenodd"
              d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
              fill-rule="evenodd"
            />
          </symbol>
          <symbol
            id="icon-bookmark"
            viewBox="0 0 17 18"
          >
            <path
              d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
            />
          </symbol>
          <symbol
            id="icon-star"
            viewBox="0 0 13 12"
          >
            <path
              clip-rule="evenodd"
              d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
              fill-rule="evenodd"
            />
          </symbol>
        </svg>
      </div>
      <div
        class="page"
      >
        <div>
          Header
        </div>
        <main
          class="page__main page__main--favorites page__main--favorites-empty"
        >
          <div
            class="page__favorites-container container"
          >
            <section
              class="favorites favorites--empty"
            >
              <h1
                class="visually-hidden"
              >
                Favorites (empty)
              </h1>
              <div
                class="favorites__status-wrapper"
              >
                <b
                  class="favorites__status"
                >
                  Nothing yet saved.
                </b>
                <p
                  class="favorites__status-description"
                >
                  Save properties to narrow down search or plan your future trips.
                </p>
              </div>
            </section>
          </div>
        </main>
        <footer
          class="footer container"
        >
          <a
            href="/"
          >
            <div
              class="footer__logo-link"
            >
              <img
                alt="6 cities logo"
                class="footer__logo"
                height="33"
                src="img/logo.svg"
                width="64"
              />
            </div>
          </a>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
